### ISSUE TITLE
First install and error

### ISSUE THREAD

**Issue by FMGordillo** [pwcremin/RNSync#2]
# Steps to reproduce error
1- `yarn add rnsync`
2- `react-native link rnsync`
3- Fixed Android dependencies (*)
4- `react-native run-android`

# build.gradle dependencies (*)

```
apply plugin: 'com.android.library'

android {
    compileSdkVersion 26
    buildToolsVersion "26.0.2"

    defaultConfig {
        minSdkVersion 19
        targetSdkVersion 26
        versionCode 1
        versionName "1.0"

        ndk {
            abiFilters "armeabi-v7a", "x86"
        }
    }
    lintOptions {
        warning 'InvalidPackage'
    }
}

repositories {
    mavenLocal()
    mavenCentral()
}

dependencies {
    compile 'com.facebook.react:react-native:+'
    compile group: 'com.cloudant', name: 'cloudant-sync-datastore-android', version:'latest.release'
    compile 'com.google.code.gson:gson:2.8.0'
}
```

# The error

**D:\Development\WatsonMobile\android> ./gradlew.bat installDebug --stacktrace**

> Configure project :app
Configuration 'compile' in project ':app' is deprecated. Use 'implementation' instead.

> Configure project :react-native-audio
Configuration 'compile' in project ':react-native-audio' is deprecated. Use 'implementation' instead.

> Configure project :react-native-watson
Configuration 'compile' in project ':react-native-watson' is deprecated. Use 'implementation' instead.

> Configure project :rnsync
Configuration 'compile' in project ':rnsync' is deprecated. Use 'implementation' instead.

> Task :rnsync:compileDebugJavaWithJavac
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java:14: error: cannot find symbol
import com.cloudant.sync.query.IndexManager;
                              ^
  symbol:   class IndexManager
  location: package com.cloudant.sync.query
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java:16: error: package com.cloudant.sync.datastore does not exist
import com.cloudant.sync.datastore.Datastore;
                                  ^
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java:17: error: package com.cloudant.sync.datastore does not exist
import com.cloudant.sync.datastore.DatastoreManager;
                                  ^
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java:18: error: package com.cloudant.sync.datastore does not exist
import com.cloudant.sync.datastore.DocumentBodyFactory;
                                  ^
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java:20: error: package com.cloudant.sync.datastore does not exist
import com.cloudant.sync.datastore.DocumentRevision;
                                  ^
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java:21: error: package com.cloudant.sync.datastore does not exist
import com.cloudant.sync.datastore.UnsavedFileAttachment;
                                  ^
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java:23: error: package com.cloudant.sync.notifications does not exist
import com.cloudant.sync.notifications.ReplicationCompleted;
                                      ^
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java:24: error: package com.cloudant.sync.notifications does not exist
import com.cloudant.sync.notifications.ReplicationErrored;
                                      ^
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java:25: error: cannot find symbol
import com.cloudant.sync.replication.ErrorInfo;
                                    ^
  symbol:   class ErrorInfo
  location: package com.cloudant.sync.replication
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java:48: error: cannot find symbol
    public ErrorInfo error = null;
           ^
  symbol:   class ErrorInfo
  location: class Listener
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java:57: error: cannot find symbol
    public void complete(ReplicationCompleted event) {
                         ^
  symbol:   class ReplicationCompleted
  location: class Listener
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java:64: error: cannot find symbol
    public void error(ReplicationErrored event) {
                      ^
  symbol:   class ReplicationErrored
  location: class Listener
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java:72: error: cannot find symbol
    private DatastoreManager manager;
            ^
  symbol:   class DatastoreManager
  location: class RNSyncModule
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java:75: error: cannot find symbol
    private Datastore ds;
            ^
  symbol:   class Datastore
  location: class RNSyncModule
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java:76: error: cannot find symbol
    private IndexManager im;
            ^
  symbol:   class IndexManager
  location: class RNSyncModule
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java:325: error: cannot find symbol
    private HashMap<String, Object> createDoc(DocumentRevision revision)
                                              ^
  symbol:   class DocumentRevision
  location: class RNSyncModule
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java:82: error: cannot find symbol
        manager = new DatastoreManager(path.getAbsolutePath());
                      ^
  symbol:   class DatastoreManager
  location: class RNSyncModule
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java:97: error: cannot find symbol
            im = new IndexManager(ds);
                     ^
  symbol:   class IndexManager
  location: class RNSyncModule
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java:183: error: cannot find symbol
        DocumentRevision revision;
        ^
  symbol:   class DocumentRevision
  location: class RNSyncModule
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java:186: error: cannot find symbol
            revision = new DocumentRevision(id);
                           ^
  symbol:   class DocumentRevision
  location: class RNSyncModule
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java:189: error: cannot find symbol
            revision = new DocumentRevision();
                           ^
  symbol:   class DocumentRevision
  location: class RNSyncModule
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java:193: error: cannot find symbol
            revision.setBody(DocumentBodyFactory.create(new HashMap<String, Object>()));
                             ^
  symbol:   variable DocumentBodyFactory
  location: class RNSyncModule
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java:196: error: cannot find symbol
            revision.setBody(DocumentBodyFactory.create(nativeBody.toHashMap()));
                             ^
  symbol:   variable DocumentBodyFactory
  location: class RNSyncModule
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java:200: error: cannot find symbol
            DocumentRevision saved = ds.createDocumentFromRevision(revision);
            ^
  symbol:   class DocumentRevision
  location: class RNSyncModule
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java:218: error: cannot find symbol
            DocumentRevision revision = ds.getDocument(id);
            ^
  symbol:   class DocumentRevision
  location: class RNSyncModule
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java:221: error: cannot find symbol
            UnsavedFileAttachment att1 =
            ^
  symbol:   class UnsavedFileAttachment
  location: class RNSyncModule
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java:222: error: cannot find symbol
                    new UnsavedFileAttachment(new File(path), type);
                        ^
  symbol:   class UnsavedFileAttachment
  location: class RNSyncModule
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java:225: error: cannot find symbol
            DocumentRevision updated = ds.updateDocumentFromRevision(revision);
            ^
  symbol:   class DocumentRevision
  location: class RNSyncModule
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java:240: error: cannot find symbol
            DocumentRevision revision = ds.getDocument(id);
            ^
  symbol:   class DocumentRevision
  location: class RNSyncModule
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java:256: error: cannot find symbol
            DocumentRevision revision = ds.getDocument(id);
            ^
  symbol:   class DocumentRevision
  location: class RNSyncModule
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java:260: error: cannot find symbol
            revision.setBody(DocumentBodyFactory.create(nativeBody.toHashMap()));
                             ^
  symbol:   variable DocumentBodyFactory
  location: class RNSyncModule
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java:262: error: cannot find symbol
            DocumentRevision updated = ds.updateDocumentFromRevision(revision);
            ^
  symbol:   class DocumentRevision
  location: class RNSyncModule
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java:278: error: cannot find symbol
            DocumentRevision revision = ds.getDocument(id);
            ^
  symbol:   class DocumentRevision
  location: class RNSyncModule
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java:315: error: cannot find symbol
        for (DocumentRevision revision : result) {
             ^
  symbol:   class DocumentRevision
  location: class RNSyncModule
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncPackage.java:15: error: method does not override or implement a method from a supertype
    @Override
    ^
Note: D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
35 errors

**FAILURE: Build failed with an exception.**

* What went wrong:
Execution failed for task ' **:rnsync:compileDebugJavaWithJavac** '.
> Compilation failed; see the compiler error output for details.

```
* Try:
Run with --info or --debug option to get more log output.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':rnsync:compileDebugJavaWithJavac'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:100)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:70)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:63)
        at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:88)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:248)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:197)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:107)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:241)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:230)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.processTask(DefaultTaskPlanExecutor.java:124)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.access$200(DefaultTaskPlanExecutor.java:80)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:105)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:99)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:625)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:580)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:99)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: org.gradle.api.internal.tasks.compile.CompilationFailedException: Compilation failed; see the compiler error output for details.
        at org.gradle.api.internal.tasks.compile.JdkJavaCompiler.execute(JdkJavaCompiler.java:51)
        at org.gradle.api.internal.tasks.compile.JdkJavaCompiler.execute(JdkJavaCompiler.java:36)
        at org.gradle.api.internal.tasks.compile.NormalizingJavaCompiler.delegateAndHandleErrors(NormalizingJavaCompiler.java:99)
        at org.gradle.api.internal.tasks.compile.NormalizingJavaCompiler.execute(NormalizingJavaCompiler.java:52)
        at org.gradle.api.internal.tasks.compile.NormalizingJavaCompiler.execute(NormalizingJavaCompiler.java:37)
        at org.gradle.api.internal.tasks.compile.CleaningJavaCompilerSupport.execute(CleaningJavaCompilerSupport.java:35)
        at org.gradle.api.internal.tasks.compile.CleaningJavaCompilerSupport.execute(CleaningJavaCompilerSupport.java:25)
        at org.gradle.api.internal.tasks.compile.incremental.IncrementalCompilationFinalizer.execute(IncrementalCompilationFinalizer.java:39)
        at org.gradle.api.internal.tasks.compile.incremental.IncrementalCompilationFinalizer.execute(IncrementalCompilationFinalizer.java:24)
        at org.gradle.api.tasks.compile.JavaCompile.performCompilation(JavaCompile.java:198)
        at org.gradle.api.tasks.compile.JavaCompile.compile(JavaCompile.java:129)
        at com.android.build.gradle.tasks.factory.AndroidJavaCompile.compile(AndroidJavaCompile.java:95)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
        at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$IncrementalTaskAction.doExecute(DefaultTaskClassInfoStore.java:173)
        at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.execute(DefaultTaskClassInfoStore.java:134)
        at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.execute(DefaultTaskClassInfoStore.java:121)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:122)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:197)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:107)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:111)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:92)
        ... 27 more


* Get more help at https://help.gradle.org
```
**BUILD FAILED in 1s**

**Comment by FMGordillo** [comment]
And compiling with Android itself, it gave me a hint:

Lots of these dependencies are referred to cloudant...

```
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncModule.java
Error:(14, 31) error: cannot find symbol class IndexManager
Error:(16, 35) error: package com.cloudant.sync.datastore does not exist
Error:(17, 35) error: package com.cloudant.sync.datastore does not exist
Error:(18, 35) error: package com.cloudant.sync.datastore does not exist
Error:(20, 35) error: package com.cloudant.sync.datastore does not exist
Error:(21, 35) error: package com.cloudant.sync.datastore does not exist
Error:(23, 39) error: package com.cloudant.sync.notifications does not exist
Error:(24, 39) error: package com.cloudant.sync.notifications does not exist
Error:(25, 37) error: cannot find symbol class ErrorInfo
Error:(48, 12) error: cannot find symbol class ErrorInfo
Error:(57, 26) error: cannot find symbol class ReplicationCompleted
Error:(64, 23) error: cannot find symbol class ReplicationErrored
Error:(72, 13) error: cannot find symbol class DatastoreManager
Error:(75, 13) error: cannot find symbol class Datastore
Error:(76, 13) error: cannot find symbol class IndexManager
Error:(325, 47) error: cannot find symbol class DocumentRevision
Error:(82, 23) error: cannot find symbol class DatastoreManager
Error:(97, 22) error: cannot find symbol class IndexManager
Error:(183, 9) error: cannot find symbol class DocumentRevision
Error:(186, 28) error: cannot find symbol class DocumentRevision
Error:(189, 28) error: cannot find symbol class DocumentRevision
Error:(193, 30) error: cannot find symbol variable DocumentBodyFactory
Error:(196, 30) error: cannot find symbol variable DocumentBodyFactory
Error:(200, 13) error: cannot find symbol class DocumentRevision
Error:(218, 13) error: cannot find symbol class DocumentRevision
Error:(221, 13) error: cannot find symbol class UnsavedFileAttachment
Error:(222, 25) error: cannot find symbol class UnsavedFileAttachment
Error:(225, 13) error: cannot find symbol class DocumentRevision
Error:(240, 13) error: cannot find symbol class DocumentRevision
Error:(256, 13) error: cannot find symbol class DocumentRevision
Error:(260, 30) error: cannot find symbol variable DocumentBodyFactory
Error:(262, 13) error: cannot find symbol class DocumentRevision
Error:(278, 13) error: cannot find symbol class DocumentRevision
Error:(315, 14) error: cannot find symbol class DocumentRevision
D:\Development\WatsonMobile\node_modules\rnsync\android\src\main\java\com\patrickcremin\react\RNSyncPackage.java
Error:(15, 5) error: method does not override or implement a method from a supertype
```

**Comment by ThomWright** [comment]
This repo was written as a wrapper around v1 of [cloudant's sync-android](https://github.com/cloudant/sync-android) project, which is now at v2 with [breaking changes](https://github.com/cloudant/sync-android/releases/tag/2.0.0). Unfortunately this repo's `build.gradle` requests `latest.version` [here](https://github.com/pwcremin/RNSync/blob/a5e80acf1deee9f06f40e3b4819e6b3f23067d66/android/build.gradle#L29), so it's using an incompatible version.

There are two possible fixes:

1. fix the version number to v1 (`1.1.5`?)
2. fix the version number to v2, using the [upgrade guide](https://github.com/cloudant/sync-android/blob/2.0.0/doc/migration.md)

Fix 1 is quick and easy, and probably all you need.

Fix 2 is better long-term, but requires more work and is probably a breaking change. I need to use v2 of the cloudant lib, so I'm planning to try this.

Either way, unless we fork we'll need the author to publish the fixes to NPM.

**Comment by pwcremin** [comment]
New NPM package 4.0 with Thom's changes
